### PR TITLE
Configure daily update

### PR DESCRIPTION
**Pull request by mweidling**
This PR adds the config for CircleCI that

- installs the project
- fetches the latest changes of the OCR-D projects specified (in `kwalitee/config.yml`)
- builds the `repos.json`
- and pushes the `repos.json` back to `main` if the file has been changed

Currently, no changes of the JSON in `repos.json` have been made; It is equivalent to the legacy specs.

Some parts of this feature are not covered by code, though. The following steps have to be taken **after** the PR has been merged:

- a new SSH key set for this project has to be created and the keys have to be added to both GitHub and CircleCI
- the fingerprint of the CircleCI key (i.e. the private key) has to be added to the `config.yml`
- the nightly trigger has to be configured in the CircleCI app

**Since this PR has a lot of debugging commits, please select the squashing option when merging.**

See https://github.com/OCR-D/zenhub/issues/97.

### CODE REVIEW

**Comment by mweidling** [comment]
@kba I still have to check the changes in `repos.json`, but feel free to have a look.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            - "submodules"
+      - run:
+          name: Update repos.json
+          command: |
+            . venv/bin/activate
+            make repos.json
+      - save_cache:
+          key: deps1-{{ .Branch }}-{{ checksum "repos.json" }}
+          paths:
+            - "repos.json"
```
Comment (kba) [inline .circleci/config.yml:36]: Why cache the `repos.json`?

**Review by kba — CHANGES_REQUESTED** [review]
LGTM, except for keeping submodules up-to-date in two places (ocrd_all and here).

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,39 @@
+[submodule "submodules/cor-asv-ann"]
```
[the remark this answers is not in the document]
Comment (mweidling) [reply .gitmodules:1]: Good idea, this also helps me with my future development.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      - add_ssh_keys:
+          fingerprints:
+            - "so:me:fin:ge:rp:ri:nt"
+      - run:
+          name: Push repos.json back to repository
+          command: |
+              git config user.email "weidling@sub.uni-goettingen.de"
```
[the remark this answers is not in the document]
Comment (mweidling) [reply .circleci/config.yml:43]: Yep, that's right. The right way to go in GitHub would be a machine user, no?

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      - image: cimg/python:3.10.1
+    steps:
+      - checkout
+      - run:
+          name: Install
+          command: |
+            python3 -m venv venv
+            . venv/bin/activate
+            pip install -r requirements.txt
+            pip install -e .
```
[the remark this answers is not in the document]
Comment (mweidling) [reply .circleci/config.yml:14]: Mmh, but `setup.py` already has a dependency in [L3](https://github.com/OCR-D/kwalitee-dashboard-back-end/blob/main/setup.py#L3):

`from ocrd_utils import VERSION`

If you install this project in an environment without ocrd already installed, a simple `pip install .` leads to the following error:

`ModuleNotFoundError: No module named 'ocrd_utils'`

File: .circleci/config.yml (hunk 1 above)
Comment (kba) [reply .circleci/config.yml:36]: AFAIK, you don't need to cache any data between steps in a single workflow. Caching is only relevant for when you want to reuse data produced in a previous run in a new run. In fact, I imagine possible cache invalidation issues if `repos.json` is cached.

File: .circleci/config.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (kba) [reply .circleci/config.yml:43]: Yes, ideally we create an `OCR-D-Bot` user or similar for these kinds of operations. Would also make sense to do the building of the website in the CI with a similar mechanism that you laid out here.

File: .circleci/config.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (mweidling) [reply .circleci/config.yml]: Ah, okay. I just updated the setup.py and made it independent from `ocrd_utils`.

File: .circleci/config.yml (hunk 1 above)
Comment (mweidling) [reply .circleci/config.yml]: Yeah, I got that wrong. My bad. Removed the caching for `repos.json`.

File: .circleci/config.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (mweidling) [reply .circleci/config.yml]: I created the user @OCR-D-automation-bot for this purpose.

File: .gitmodules (hunk 2 above)
[the remark this answers is not in the document]
Comment (mweidling) [reply .gitmodules]: Done, see [1a716ed](https://github.com/OCR-D/kwalitee-dashboard-back-end/pull/4/commits/1a716edf6e11d53a6d52262476af622cd7582f41) and [e2ecbdd](https://github.com/OCR-D/kwalitee-dashboard-back-end/pull/4/commits/e2ecbdd618e10c19a1866bda05ca726254ce00d5).

**Comment by mweidling** [comment]
@kba At this point i'd remove the "Draft" status of the PR and suggest to merge it. Maybe some fixing will be necessary with the bot mechanism, but it is easier to try this out in this repository instead of the fork.

If you don't have any objections, feel free to review my changes and merge.

**Comment by mweidling** [comment]
The failing check is a result of the bot not being allowed to push to my fork (in case you're wondering).

**Pull request merged**